File: solved.ac/code/9461.js
const fs = require("fs")
const input = fs.readFileSync("./dev/stdin").toString().trim().split("\n")
const n = input.shift()
const a = []
a[0] = 0
a[1] = 1
a[2] = 1
a[3] = 1
// console.log(a)
for (let i = 4; i <= 100; i++) {
  a[i] = a[i - 2] + a[i - 3]
}
for (let i = 0; i < n; i++) {
  console.log(a[input[i]])
}

File: solved.ac/code/9461.py
# p(1) =1
# p(2) =1
# p(3) = 1
# p(4) = 2
# p(5) = 2
# p(6) = 3
# p(7) = 4

import sys
input = sys.stdin.readline
t = int(input())
dp = [0] * 101
dp[0] = 1
dp[1] = 1
dp[2] = 1
dp[3] = 2
dp[4] = 2
dp[5] = 3


for _ in range(t):
  n = int(input())
  if dp[n] == 0 :
    for i in range(6, n + 1):
      dp[i] = dp[i-1] + dp[i-5]
  print(dp[n-1])
